### PR TITLE
Updates for python wrapper adding support for revocation status list changes

### PR DESCRIPTION
**Pull request by dbluhm**
Opening this early. This PR updates the python wrapper to accommodate changes made at the rust layer for revocation status lists.

Feedback and corrections welcome. I'll continue working on this by updating the `test.py` file to bring it in line with the `anoncreds_demo.rs`.

### CODE REVIEW

**Comment by andrewwhitehead** [comment]
You might want to integrate these changes as well? https://github.com/hyperledger/indy-shared-rs/pull/26

**Comment by dbluhm** [comment]
> You might want to integrate these changes as well? [hyperledger/indy-shared-rs#26](https://github.com/hyperledger/indy-shared-rs/pull/26)

That seems like a good idea! I'll check it out.

**Comment by blu3beri** [comment]
Awesome work! There was a slight misunderstanding on my side related to this work, I also made a draft for it as I was stuck on some other work. #146 is the related PR.

Maybe there are some things that you can take from my PR, but I will abandon mine so the work can continue here.

**Comment by dbluhm** [comment]
Shoot, apologies for the miscommunication -- duplicating work is less than ideal. It looks like you got further along than I did. I'll see what I can do to combine efforts here! Thanks for the heads up, @blu3beri

**Comment by dbluhm** [comment]
@blu3beri After combining everything, `demo/test.py` is currently failing with the following error:

```
Traceback (most recent call last):
  File "/home/dbluhm/dev/anoncreds/anoncreds-rs/wrappers/python/demo/test.py", line 192, in <module>
    verified = presentation.verify(
               ^^^^^^^^^^^^^^^^^^^^
  File "/home/dbluhm/dev/anoncreds/anoncreds-rs/wrappers/python/anoncreds/types.py", line 593, in verify
    return bindings.verify_presentation(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dbluhm/dev/anoncreds/anoncreds-rs/wrappers/python/anoncreds/bindings.py", line 803, in verify_presentation
    do_call(
  File "/home/dbluhm/dev/anoncreds/anoncreds-rs/wrappers/python/anoncreds/bindings.py", line 404, in do_call
    raise get_current_error(True)
anoncreds.error.AnoncredsError: Revocation Registry not provided for ID and timestamp: RevocationRegistryId("mock:uri:revregid"), 12
```

To see if this was caused by a change I introduced, I checked out 4a019cbee6a6b1093d60926a806bd5f6cf940423 from your branch and had the same issue. Is this expected?

**Comment by dbluhm** [comment]
The python wrapper demo is now functioning as expected; I think this is ready for a review now. Thanks for all the feedback up to this point already :slightly_smiling_face:

File: src/ffi/revocation.rs [hunk 1]
```diff
@@ -75,14 +75,14 @@ pub extern "C" fn anoncreds_update_revocation_status_list(
...
             None
+        } else {
+            Some(revoked.as_slice().iter().map(|r| *r as u32).collect())
         };
         let issued: Option<BTreeSet<u32>> = if issued.is_empty() {
-            Some(issued.as_slice().iter().map(|r| *r as u32).collect())
-        } else {
             None
+        } else {
+            Some(issued.as_slice().iter().map(|r| *r as u32).collect())
```
Comment (dbluhm) [inline src/ffi/revocation.rs:85]: I believe this condition was flipped around; if the revoked list was empty, return the list mapped to a BTreeSet<u32>, else, return None. So we were always getting back empty revoked lists.

File: wrappers/python/demo/test.py [hunk 2]
```diff
@@ -13,139 +13,263 @@
...
+
+verified = presentation.verify(
+    pres_req,
+    schemas,
+    cred_defs,
+    rev_reg_defs,
+    rev_status_lists,
+)
+
+print(verified)
```
Comment (blu3beri) [inline wrappers/python/demo/test.py:270]: This can be removed.

**Review by blu3beri — APPROVED** [review]
LGTM! Might be good is someone from the ACA-Py team gives this a review? Maybe @andrewwhitehead ?

File: src/ffi/revocation.rs (hunk 1 above)
Comment (whalelephant) [reply src/ffi/revocation.rs:85]: @dbluhm @blu3beri you are correct!

**Pull request merged**